sfdc/modules/bulk/tests/bulk_csv_insert.bal

```diff
@@ -25,146 +25,147 @@ decimal delayInSecs = 5.0;
 @test:Config {
     enable: true
 }
-function insertCsv() {
+function insertCsv() returns error? {
     log:printInfo("baseClient -> insertCsv");
     string batchId = "";
     string contacts = "description,FirstName,LastName,Title,Phone,Email,My_External_Id__c\n"
         + "Created_from_Ballerina_Sf_Bulk_API,Cuthbert,Binns,Professor Level 02,0332236677,john434@gmail.com,845\n"
         + "Created_from_Ballerina_Sf_Bulk_API,Burbage,Shane,Professor Level 02,0332211777,peter77@gmail.com,846";
 
     //create job
-    error|BulkJob insertJob = baseClient->createJob("insert", "Contact", "CSV");
+    BulkJob insertJob = check baseClient->createJob("insert", "Contact", "CSV");
 
-    if (insertJob is BulkJob) {
-        //add csv content
-        foreach var i in 1 ..< maxIterations {
-            error|BatchInfo batch = baseClient->addBatch(insertJob, contacts);
-            if (batch is BatchInfo) {
-                test:assertTrue(batch.id.length() > 0, msg = "Could not upload the contacts using CSV.");
-                batchId = batch.id;
-                break;
+    //add csv content
+    foreach var i in 1 ..< maxIterations {
+        error|BatchInfo batch = baseClient->addBatch(insertJob, contacts);
+        if (batch is BatchInfo) {
+            test:assertTrue(batch.id.length() > 0, msg = "Could not upload the contacts using CSV.");
+            batchId = batch.id;
+            break;
+        } else {
+            if i != 5 {
+                log:printWarn("addBatch Operation Failed! Retrying...");
+                runtime:sleep(delayInSecs);
             } else {
-                if i != 5 {
-                    log:printWarn("addBatch Operation Failed! Retrying...");
-                    runtime:sleep(delayInSecs);
-                } else {
-                    log:printWarn("addBatch Operation Failed! Giving up...");
-                    test:assertFail(msg = batch.message());
-                }
+                log:printWarn("addBatch Operation Failed! Giving up...");
```

Comment:
Giving up after x tries?

Reply:
Fixed in https://github.com/ballerina-platform/module-ballerinax-sfdc/commit/947cabc64eb9743b7e8c7e0d0d234c46844f11b2